src/kudu/consensus/consensus_queue.cc

```diff
@@ -883,6 +887,97 @@ void PeerMessageQueue::AdvanceQueueWatermark(const char* type,
+      const string& peer_region = peer.second->peer_pb.attrs().region();
+      std::vector<int64_t>& regional_watermarks = LookupOrInsert(
+          &watermarks_by_region, peer_region, std::vector<int64_t>());
+      regional_watermarks.push_back(peer.second->last_received.index());
+    }
+  }
+
+  // Vector of maximum indexes which satisfy commit requirements in each region.
+  std::vector<int64_t> commit_index_regional_level;
+  for (const std::pair<std::string, int>& regional_commit_requirement :
+      data_commit_quorum) {
+    std::map<std::string, std::vector<int64_t> >::iterator it =
+        watermarks_by_region.find(regional_commit_requirement.first);
+    if (it == watermarks_by_region.end() ||
+        it->second.size() < regional_commit_requirement.second) {
+      VLOG_WITH_PREFIX_UNLOCKED(3)
+          << "Watermarks size: "
+          << ((it == watermarks_by_region.end()) ? 0 : it->second.size())
+          << ", Num peers required: " << regional_commit_requirement.second
+          << ", Region: " << regional_commit_requirement.first;
+      return;
+    }
+    std::sort(it->second.begin(), it->second.end());
+    commit_index_regional_level.push_back(
+        it->second[it->second.size() - regional_commit_requirement.second]);
+  }
+
+  // Since `data_commit_requirement` corresponds to a conjunction requirement,
+  // we compute the minimum index which is considered replicated in all regions.
+  std::vector<int64_t>::iterator new_watermark_it = std::min_element(
+      commit_index_regional_level.begin(), commit_index_regional_level.end());
+  int64_t old_watermark = *watermark;
+  *watermark = *new_watermark_it;
+
+  VLOG_WITH_PREFIX_UNLOCKED(1) << "Updated " << type << " watermark "
+      << "from " << old_watermark << " to " << (*new_watermark_it);
+  if (VLOG_IS_ON(3)) {
+    VLOG_WITH_PREFIX_UNLOCKED(3) << "Peers: ";
+    for (const PeersMap::value_type& peer : peers_map_) {
+      VLOG_WITH_PREFIX_UNLOCKED(3) << "Peer: " << peer.second->ToString();
+    }
+    VLOG_WITH_PREFIX_UNLOCKED(3) << "Data commit quorum: ";
+    for (const std::pair<std::string, int>& regional_commit_requirement :
+        data_commit_quorum) {
+      VLOG_WITH_PREFIX_UNLOCKED(3)
+          << "Region: " << regional_commit_requirement.first << ", "
+          << "Count: " << regional_commit_requirement.second;
+    }
+    VLOG_WITH_PREFIX_UNLOCKED(3) << "Sorted watermarks:";
+    for (const std::pair<std::string, vector<int64_t> >& region_watermarks :
+        watermarks_by_region) {
+      VLOG_WITH_PREFIX_UNLOCKED(3) << "Region: " << region_watermarks.first;
+      for (const int64_t watermark : region_watermarks.second) {
+        VLOG_WITH_PREFIX_UNLOCKED(3) << "Watermark: " << watermark;
+      }
+    }
+  }
+}
+
+std::map<std::string, int> PeerMessageQueue::GetDataCommitQuorum() {
+  std::map<std::string, int> data_commit_quorum;
+  // TODO(ritwikyadav) : Populate quorum from cmeta_
+  return data_commit_quorum;
```

Comment:
for the sake of the immediate needs, my suggestion would be to have

DCQ = { Majority of voters in master Region }
LEQ = { Dual of it, which is Majority of voters in all regions (conjunction) }

This will give us more time to work towards the real deal.
DCQ = { Majority of voters in Master Region }
LEQ = { Intersect with the latest DCQ }